java/code/src/com/redhat/rhn/manager/channel/ChannelManager.java

```diff
@@ -2558,6 +2558,10 @@ public static Optional<DataResult<EssentialChannelDto>> listPossibleSuseBaseChan
         return Opt.fold(baseProductId,
                 () -> {
                     log.info("Server has no base product installed");
+                    if (s.getBaseChannel() != null) {
```

Comment:
you can simplify that a bit writing it like this

```java
     return ofNullable(s.getBaseChannel())
                            .map(bs -> new DataResult(List.of(new EssentialChannelDto(bs))));
```

Reply:
This has been changed to a more complex `if` statement already.

Reply:
I leave this now, as the other branches have already been merged. We can change it later if needed.